File: 2017-multi-jpa-java/core/src/main/java/sample/member/core/transfer/PageResponseWrapper.java
package sample.member.core.transfer;

import lombok.Data;

import java.util.List;

@Data
public class PageResponseWrapper<RESPONSE> {

	public static <RESPONSE0> PageResponseWrapper<RESPONSE0> create() {
		return new PageResponseWrapper<>();
	}

	private boolean success = true;

	public PageResponseWrapper<RESPONSE> success(boolean success) {
		this.success = success;
		return this;
	}

	private String lang;

	public PageResponseWrapper<RESPONSE> lang(String lang) {
		this.lang = lang;
		return this;
	}

	private String code;

	public PageResponseWrapper<RESPONSE> code(String code) {
		this.code = code;
		return this;
	}

	private String message;

	public PageResponseWrapper<RESPONSE> message(String message) {
		this.message = message;
		return this;
	}

	private int totalPages;

	public PageResponseWrapper<RESPONSE> totalPages(int totalPages) {
		this.totalPages = totalPages;
		return this;
	}

	private long totalElements;

	public PageResponseWrapper<RESPONSE> totalElements(long totalElements) {
		this.totalElements = totalElements;
		return this;
	}

	private List<RESPONSE> data;

	public PageResponseWrapper<RESPONSE> data(List<RESPONSE> data) {
		this.data = data;
		return this;
	}
}

File: 2021-rabbitmq-jpa/core/src/main/kotlin/sample/core/transfer/PageResponseWrapper.kt
package sample.core.transfer

class PageResponseWrapper<RESPONSE> {
    private var success = true
    fun success(success: Boolean): PageResponseWrapper<RESPONSE> {
        this.success = success
        return this
    }

    private var lang: String? = null
    fun lang(lang: String?): PageResponseWrapper<RESPONSE> {
        this.lang = lang
        return this
    }

    private var code: String? = null
    fun code(code: String?): PageResponseWrapper<RESPONSE> {
        this.code = code
        return this
    }

    private var message: String? = null
    fun message(message: String?): PageResponseWrapper<RESPONSE> {
        this.message = message
        return this
    }

    private var totalPages = 0
    fun totalPages(totalPages: Int): PageResponseWrapper<RESPONSE> {
        this.totalPages = totalPages
        return this
    }

    private var totalElements: Long = 0
    fun totalElements(totalElements: Long): PageResponseWrapper<RESPONSE> {
        this.totalElements = totalElements
        return this
    }

    private var data: List<RESPONSE>? = null
    fun data(data: List<RESPONSE>?): PageResponseWrapper<RESPONSE> {
        this.data = data
        return this
    }

    companion object {
        fun <RESPONSE0> create(): PageResponseWrapper<RESPONSE0> {
            return PageResponseWrapper()
        }
    }
}
